javalib/src/main/scala/java/util/Timer.scala

```diff
@@ -0,0 +1,115 @@
+package java.util
+
+import scala.collection._
+import scala.concurrent.duration._
+import scala.scalajs.js.timers._
+
+class Timer(val name: String, val isDaemon: Boolean) {
+  private[util] var cancelled = false
+  private[util] val tasks = mutable.Set[TimerTask]()
+
+  def this() = this("timer", false)
+
+  def this(isDaemon: Boolean) = this("timer", isDaemon)
+
+  def this(name: String) = this(name, false)
+
+  private[util] def scheduleOnce(task: TimerTask, delay: Long): Unit = {
+    tasks += task
+    setTimeout(delay.millis) {
+      task.doRun()
+    }
+  }
+
+  private[util] def getMillisUntil(time: Date): Long = {
+    math.max(0, time.getTime - (new Date()).getTime)
```

Comment:
Consider using `System.currentTimeMillis()` instead of `(new Date()).getTime`, which avoids the allocation of the `java.util.Date` object.